### PR TITLE
Initial refactor of multi select

### PR DESCRIPTION
**Pull request by AmirAsaraf**
This PR includes the following:

1) React-shared list to list import to this repo
2) Remove all shared dependencies.
3) Use material in checkbox, input and loader
4) Use specific colors in css
5) add waypoint and react-virtualized libs
6) Add s sample in stories
7) Refactoring of names and code

**Not includes:**
1) scss refactor and fixes
2) tests
3) bug fixes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +71,8 @@
     "webpack-cli": "^2.0.12",
     "webpack-dev-server": "^2.2.0",
     "webpack-merge": "^4.1.2",
-    "webpack-strip": "^0.1.0"
+    "webpack-strip": "^0.1.0",
+    "lodash": "latest"
```
Comment (liorheber) [inline package.json]: Lets avoid latest tags and use the current latest as the version used.

Comment (liorheber) [reply package.json]: Also, I think it should be in dependencies if you're using it in the component.

File: src/index.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
-import MultiSelectionList from './multiselection_list';
+import ReactMultiSelect from "./react_multi_select";
```
Comment (liorheber) [inline src/index.js]: export { default } from "./react_multi_select"

File: src/multi_selection_list/multiselection_list.scss [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
+$item_font_color: #666666;
 $item_padding: 10px;
 $grouped_item_padding: 20px;
 $item_height: 30px;
 $list_height: 218px;
 $navigation_list_height: 178px;
 $list_width: 250px;
 $list_font_size: 12px;
-$list_error_color: $palette_red_300;
+$list_error_color: #FF3333;
```
Comment (liorheber) [inline src/multi_selection_list/multiselection_list.scss]: Do you think we should let go of some minimal scss file with the defaults?

**Comment by liorheber** [comment]
👍

File: package.json (hunk 1 above)
Comment (AmirAsaraf) [reply package.json]: ok

File: src/index.js (hunk 2 above)
Comment (AmirAsaraf) [reply src/index.js]: sure

**Comment by AmirAsaraf** [comment]
For now, It looks a bit ugly. I will keep work on the styling to make it look good

File: src/multi_selection_list/multiselection_list.scss (hunk 3 above)
Comment (liorheber) [reply src/multi_selection_list/multiselection_list.scss]: Yeah, I don't think we need to deal with it in this PR.

**Pull request merged**